src/core/items/ieffects.ts

```diff
@@ -764,4 +803,83 @@ export class ItemEffect implements IItemEffect {
       })
     });
   }
+
+  getFilter(): Promise<Filter> {
+    return new Promise(resolve => {
+      iItem.get('prop:effects', this._id)
+      .then(val => {
+        try {
+          var effectsJXON = JXON.parse(val);
+          resolve(new Filter(effectsJXON['children'][0]['id']));
+        } catch(e) {
+          resolve(Filter.NONE);
+        }
+      });
+    });
+  }
+
+  setFilter(value: any, config ?: {
+    intensity ?: number,
+    resourceFile ?: string
+  }): Promise<ItemEffect> {
+    return new Promise((resolve, reject) => {
```

Comment:
Yes, there was a previous task from way back to standardize our methods to just always use a promise. I can't seem to find it though. I will also file another issue to check all set methods, because it seems some of them have no resolutions.